### PR TITLE
Nxengine

### PR DESCRIPTION
**Pull request by Yonle**

### CODE REVIEW

**Review by WMCB-Tech — CHANGES_REQUESTED** [review]
~~use `@TERMUX_PREFIX@` instead of `$PREFIX`~~

~~commit:
https://github.com/termux/x11-packages/pull/379/commits/911edf70a97710446c132946c373ee6a27d5330a~~

Have you done this, cuz i saw your commits using `$PREFIX` instead of `@TERMUX_PREFIX` in your patchfile

File: packages/nxengine/build.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+TERMUX_PKG_HOMEPAGE=https://nxengine.sourceforge.net
+TERMUX_PKG_DESCRIPTION="Open-source rewrite engine of the Cave Story for Dingux and MotoMAGX. Author - Caitlin Shaw (rogueeve), httpd://nxengine.sourceforge.net"
```
Comment (xeffyr) [inline packages/nxengine/build.sh:2]: No **long** descriptions. Keep description short, but informative. There no need to keep author and homepage in description, btw the latter one is stored in a separate field `TERMUX_PKG_HOMEPAGE`.

File: packages/nxengine/build.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+TERMUX_PKG_LICENSE="GPL-3.0"
+TERMUX_PKG_MAINTAINER="@Yonle"
+TERMUX_PKG_VERSION=1.0.0.4
+TERMUX_PKG_REVISION=4
```
Comment (xeffyr) [inline packages/nxengine/build.sh:6]: Why revision starts from 4? This is a new package and therefore it shouldn't have revision field set. Revision is a build number and incremented each time when package should be rebuilt.

File: packages/nxengine/build.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+TERMUX_PKG_SRCURL=https://github.com/EXL/NXEngine/archive/refs/tags/v1.0.0.4-Rev4.tar.gz
```
Comment (xeffyr) [inline packages/nxengine/build.sh:7]: Why version is hardcoded in srcurl? Reference it from `TERMUX_PKG_VERSION`.

**Comment by Yonle** [comment]
# GOD
WHAT THE HELL JUST HAPPEND, THIS IS NOT READY YET. DID I MISSCLICK!?

File: packages/nxengine/fix-prefix.patch [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+@@ -16,11 +16,11 @@
+ CXX           = g++
+ DEFINES       = # -D_L10N_CP1251
+ CFLAGS        = -pipe -O2 $(DEFINES)
+ CXXFLAGS      = -pipe -O2 $(DEFINES)
+-INCPATH       = -I/usr/include -I/usr/include/SDL -I/usr/local/include -I/usr/local/include/SDL -I.
++INCPATH       = -I$TERMUX_PREFIX/include -I$TERMUX_PREFIX/include/SDL -I$TERMUX_PREFIX/local/include -I$TERMUX_PREFIX/local/include/SDL -I.
+ LINK          = g++
+-LFLAGS        = -Wl,-O1 -Wl,-rpath,/usr/lib
++LFLAGS        = -Wl,-O1 -Wl,-rpath,$TERMUX_PREFIX/lib
```
Comment (xeffyr) [inline packages/nxengine/fix-prefix.patch:13]: Why not to just use flags from LDFLAGS? Similar issue with INCPATH...

**Comment by Yonle** [comment]
Since i'm not sure, This may be complete. 

So sorry about this. This is really a mistake.

File: packages/nxengine/fix-prefix.patch [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+ LIBS_SHARED   = -lSDL -lSDL_ttf
+-LIBS_STATIC   = /usr/local/lib/libSDLmain.a /usr/local/lib/libSDL.a /usr/local/lib/libSDL_ttf.a -ldl -lpthread -lfreetype
++LIBS_STATIC   = $TERMUX_PREFIX/local/lib/libSDLmain.a $TERMUX_PREFIX/local/lib/libSDL.a $TERMUX_PREFIX/local/lib/libSDL_ttf.a -ldl -lpthread -lfreetype
```
Comment (xeffyr) [inline packages/nxengine/fix-prefix.patch:16]: Why there is a `/local` path component? Remove it. It is not supported officially and shouldn't be used in any of our packages.

If you need it for personal purposes, keep such changes in your fork instead.

**Review by xeffyr — CHANGES_REQUESTED** [review]
Way to go.
* https://github.com/termux/termux-packages/wiki/Coding-guideline
* https://github.com/termux/termux-packages/wiki/Creating-new-package#writing-a-package-script

**Comment by xeffyr** [comment]
if you will want to re-submit a PR, please check the issues I've pointed in review comments.

File: packages/nxengine/fix-prefix.patch (hunk 4 above)
Comment (xeffyr) [reply packages/nxengine/fix-prefix.patch:13]: P.S. Compilers and other related vars should be retrieved from environment variables too.

**Comment by Yonle** [comment]
By the way, I'm gonna resubmit the PR later. I may need to fix the building issue so i should fix it in my fork. Once i finish it, Gonna re-submit PR again. 

Also thanks for the issues. I'm gonna fix that sooner in next PR.

**Pull request closed without merging**